internal/usecases/liquidity_provider/summaries.go

```diff
@@ -0,0 +1,220 @@
+package liquidity_provider
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/rsksmart/liquidity-provider-server/internal/entities"
+	"github.com/rsksmart/liquidity-provider-server/internal/entities/quote"
+	"github.com/rsksmart/liquidity-provider-server/internal/usecases"
+)
+
+const (
+	DateFormat = time.DateOnly
+)
+
+type SummaryResult struct {
+	PeginSummary  SummaryData `json:"peginSummary"`
+	PegoutSummary SummaryData `json:"pegoutSummary"`
+}
+
+type SummaryData struct {
+	TotalQuotesCount          int64         `json:"totalQuotesCount"`
+	AcceptedQuotesCount       int64         `json:"acceptedQuotesCount"`
+	PaidQuotesCount           int64         `json:"paidQuotesCount"`
+	PaidQuotesAmount          *entities.Wei `json:"paidQuotesAmount"`
+	TotalAcceptedQuotedAmount *entities.Wei `json:"totalAcceptedQuotedAmount"`
+	TotalFeesCollected        *entities.Wei `json:"totalFeesCollected"`
+	RefundedQuotesCount       int64         `json:"refundedQuotesCount"`
+	TotalPenaltyAmount        *entities.Wei `json:"totalPenaltyAmount"`
+	LpEarnings                *entities.Wei `json:"lpEarnings"`
+}
+
+type SummariesUseCase struct {
+	peginRepo  quote.PeginQuoteRepository
+	pegoutRepo quote.PegoutQuoteRepository
+}
+
+func NewSummaryData() SummaryData {
+	return SummaryData{
+		TotalQuotesCount:          0,
+		AcceptedQuotesCount:       0,
+		PaidQuotesCount:           0,
+		PaidQuotesAmount:          entities.NewWei(0),
+		TotalAcceptedQuotedAmount: entities.NewWei(0),
+		TotalFeesCollected:        entities.NewWei(0),
+		RefundedQuotesCount:       0,
+		TotalPenaltyAmount:        entities.NewWei(0),
+		LpEarnings:                entities.NewWei(0),
+	}
+}
+
+func NewSummariesUseCase(peginRepo quote.PeginQuoteRepository, pegoutRepo quote.PegoutQuoteRepository) *SummariesUseCase {
+	return &SummariesUseCase{peginRepo: peginRepo, pegoutRepo: pegoutRepo}
+}
+
+func (u *SummariesUseCase) Run(ctx context.Context, startDate, endDate time.Time) (SummaryResult, error) {
+	peginData, err := u.aggregatePeginData(ctx, startDate, endDate)
+	if err != nil {
+		return SummaryResult{}, usecases.WrapUseCaseError(usecases.SummariesUseCaseId, err)
+	}
+	pegoutData, err := u.aggregatePegoutData(ctx, startDate, endDate)
+	if err != nil {
+		return SummaryResult{}, usecases.WrapUseCaseError(usecases.SummariesUseCaseId, err)
+	}
+	return SummaryResult{PeginSummary: peginData, PegoutSummary: pegoutData}, nil
+}
+
+func processPeginRetainedQuote(
+	ctx context.Context,
+	repo quote.PeginQuoteRepository,
+	retained quote.RetainedPeginQuote,
+	data *SummaryData,
+	acceptedTotalAmount, totalFees, callFees, totalPenalty *entities.Wei,
+) {
+	q, err := repo.GetQuote(ctx, retained.QuoteHash)
```

Comment:
You should query the db with the date range and get the quotes + the retained quotes in one query, because this will generate 1 read per execution, if we're querying the quotes of a month this could break the server or db